src/GraphQL/Types/Scalars/TimeSpanSecondsGraphType.cs

```diff
@@ -45,5 +44,8 @@ public TimeSpanSecondsGraphType()
             long l => TimeSpan.FromSeconds(l),
             _ => null
         };
+
+        /// <inheritdoc/>
+        public override IValue ToAST(object value) => new LongValue(Convert.ToInt64(Serialize(value)));
```

Comment:
Bug + boxing in `Serialize`. Should be
```
  public override object Serialize(object value) => value switch
        {
            TimeSpan timeSpan => (long)timeSpan.TotalSeconds,
            int i => (long)i,
            long l => value, // no boxing
            _ => null
        };
```

Please also check other `Serialize` methods to return only desired type, i.e. **only** `long` or **only** uint, etc. + "native" case for this type should return `value` to eliminate boxing.

Reply:
Also in `ParseValue` : ` TimeSpan t => t,` -> ` TimeSpan t => value, // no boxing` and so on for each scalar type.

Reply:
https://github.com/graphql-dotnet/graphql-dotnet/pull/2321/files#r584371834

Reply:
Not sure what this is pointing to.

Reply:
I think this conversation is resolved - ?

Reply:
yes